### PR TITLE
feat: Added nextjs-docs extension

### PR DESCRIPTION
**Pull request by PrasadChinwal**
## Description

Extension to read the [NextJS docs](https://nextjs.org/docs/getting-started) from Ray.

## Commands
- NextJS docs
  - Lists the various topics in NextJS docs from Github

## Configuration
- Requires a Github Personal Access token since Github API's have rate limiting.

## Screencast

nextjs-docs

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by thomaspaulmann** [comment]
Can we get this working without a GitHub API token? Feels a lil weird that you need it for such an extension. 

Instead of fetching the documentation all the time when you open the search, can we cache it? Or just bundle it while building the extension? I think we do this already for other documentation extensions, e.g. [Tailwind](https://www.raycast.com/vimtor/tailwindcss) or [Gitignore](https://www.raycast.com/daniel-stoneuk/gitignore). This would make it also faster to use the extension.

**Comment by PrasadChinwal** [comment]
@thomaspaulmann Thanks for the insights. I will submit another PR as soon as possible.

**Comment by thomaspaulmann** [comment]
@PrasadChinwal let's make it part of this PR.

Also, while you there, could you please use [the official Next.js assets](https://vercel.com/design/brands#next-js)? I think the  symbol works quite nicely and you can use [light/dark ones](https://developers.raycast.com/information/manifest) to make it look stunning in both our themes.

**Review by thomaspaulmann — CHANGES_REQUESTED** [review]
I put it back in your queue to remove the restriction of a GH API token. Let me know if you need anything. Feel free to ping me in our [community](https://raycast.com/community).

**Comment by PrasadChinwal** [comment]
Hey @thomaspaulmann, 

I have implemented the suggested changes.
- Github Personal Access Token dependency has been removed.
- Used the official NextJS assets.

Thanks for the advice!

**Review by PrasadChinwal — COMMENTED** [review]
- Removed Github Personal Access Token dependency
- Used official NextJS assets

**Review by thomaspaulmann — CHANGES_REQUESTED** [review]
Thanks a lot. Few minor things to do before merging:
1. Can you please swap the icons? The white icon should be used for the dark theme.
2. Can you please crop your [screenshots](https://developers.raycast.com/basics/prepare-an-extension-for-store#guides-and-template) to fit our guidelines? We also have a Figma template for it.
3. A small improvement: I think you can always use the cached docs first. If some treshold is passed (e.g. 24 hours) also fetch the docs and update the cache. I think the docs don't change that often so you have most of the time the cached version ready to go. This makes it faster to use the extension. We do something similar in the [Figma File Search](https://github.com/raycast/extensions/tree/1da89c4626dd39b3c50a12208ee14c816f918ae8/extensions/figma-files/).

**Comment by PrasadChinwal** [comment]
Hey @thomaspaulmann ,
I have implemented the caching of the docs in my previous PR (threshold was 24 hours). I just pushed new changes to address the below issues:
- Swapped the icons for light/dark mode.
- Did my best to follow the guidelines for the screenshot.

Thanks for going through this. Appreciate your help!

File: extensions/nextjs-docs/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "$schema": "https://www.raycast.com/schemas/extension.json",
   "name": "nextjs-docs",
-  "title": "nextjs-docs",
+  "title": "Nextjs-docs",
```
Comment (thomaspaulmann) [inline extensions/nextjs-docs/package.json:4]:
```suggestion
  "title": "Next.js Documentation",
```

File: extensions/nextjs-docs/package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
   "description": "Efficiently search NextJS documentation from RayCast.",
```
Comment (thomaspaulmann) [inline extensions/nextjs-docs/package.json:5]:
```suggestion
  "description": "Efficiently search Next.js documentation",
```

**Comment by thomaspaulmann** [comment]
> I have implemented the caching of the docs in my previous PR (threshold was 24 hours).

I saw that but AFAIK right now, it checks the cache and after 24 hours ignores the cache and refetches from the network. I would suggest the following logic:
1. Get results from cache and update list
2. Check if 24 hours passed, if yes make a network request and update the list
3. Store the updated cache if necessary

This way you always get a fast cached result. Feel free to implement that in a separate PR.

File: extensions/nextjs-docs/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        <List.Item
+          key={topic.sha}
+          icon={Icon.TextDocument}
+          title={topic.title}
+          actions={
+            <ActionPanel>
+              <Action.Push title={`Browse ${topic.title}`} target={<TopicDetail topic={topic} />} />
+            </ActionPanel>
+          }
+          accessories={[{ text: topic.filepath, icon: Icon.Finder }]}
```
Comment (thomaspaulmann) [inline extensions/nextjs-docs/src/index.tsx:39]: I would suggest not setting an icon here. The `Icon.Finder` doesn't really fit here.

Comment (thomaspaulmann) [inline extensions/nextjs-docs/src/index.tsx:39]: I would suggest not setting an icon here. The `Icon.Finder` doesn't really fit here.

```suggestion
          accessories={[{ text: topic.filepath }]}
```

**Comment by thomaspaulmann** [comment]
Can you please update the screenshots one more time? I did some smaller visual tweaks and the authentication isn't required anymore and therefore outdated. Afterwards this should be ready to go. Sorry for the back and forth...

**Comment by PrasadChinwal** [comment]
> Can you please update the screenshots one more time? I did some smaller visual tweaks and the authentication isn't required anymore and therefore outdated. Afterwards this should be ready to go. Sorry for the back and forth...

Do you mean the screenshot for the Authentication page? That screenshot shows the how the Authentication docs from NextJS. I can swap it with some other page example.

File: extensions/nextjs-docs/src/index.tsx (hunk 3 above)
Comment (PrasadChinwal) [reply extensions/nextjs-docs/src/index.tsx]: Thanks @thomaspaulmann. :)

File: extensions/nextjs-docs/package.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    "Productivity",
+    "Documentation",
+    "Developer Tools"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "NextJS Docs",
+      "description": "Search NextJS docs in Ray",
```
Comment (thomaspaulmann) [inline extensions/nextjs-docs/package.json:18]:
```suggestion
      "title": "Search Next.js Documentation",
      "description": "Search the Next.js documentation",
```

**Comment by thomaspaulmann** [comment]
Sorry, I got confused. I changed the command name to fit Next.js official naming convention.

**Comment by thomaspaulmann** [comment]
One thing you might want to do, is sorting the pages better. You could also use `List.Section` to group things like `Advanced Features` and then have in the accessories only the "sub category". Just an idea.

**Comment by PrasadChinwal** [comment]
Hey @thomaspaulmann,

Thank you so much for all the help. I'll definitely look into changing the caching logic as you mentioned, I think it is much better that way.

**Comment by thomaspaulmann** [comment]
Awesome, feel free to ping me on the PR since I have context.

**Pull request merged**